src/encord_active/lib/project/project_file_structure.py

```diff
@@ -1,18 +1,88 @@
+from __future__ import annotations
+
 import json
+import re
 from pathlib import Path
-from typing import Iterator, NamedTuple, Optional, Union
+from typing import Any, Iterator, List, Optional, Union
 
 from prisma import models
+from prisma.types import (
+    DataUnitCreateInput,
+    DataUnitInclude,
+    DataUnitWhereInput,
+    LabelRowCreateInput,
+)
 
-from encord_active.lib.db.base import DataUnitLike
-from encord_active.lib.db.compatibility import fill_missing_tables
 from encord_active.lib.db.connection import PrismaConnection
 from encord_active.lib.file_structure.base import BaseProjectFileStructure
 
 
-class DataUnitStructure(NamedTuple):
-    hash: str
-    path: Path
+# To be deprecated when Encord Active version is >= 0.1.60.
+def _fill_missing_tables(pfs: ProjectFileStructure):
+    # Adds the content missing from the data units table when projects with
+    # older versions of Encord Active are handled with versions greater than 0.1.52.
+    label_row_meta = json.loads(pfs.label_row_meta.read_text(encoding="utf-8"))
+    with PrismaConnection() as conn:
+        fill_label_rows = conn.labelrow.count() == 0
+        fill_data_units = conn.dataunit.count() == 0
+        if not fill_label_rows or fill_data_units:
+            return
+
+        with conn.batch_() as batcher:
+            for label_row in pfs.iter_labels():
```

Comment:
The plan is to deprecate `iter_labels()`, but I guess it's fine here and we will remove the compatibility code at the same time as that method.

Reply:
same for `iter_data_unit()`.

Reply:
Well, the entire thing is based on using that to transform it into the new structure. I think that they will be removed together these things.

Reply:
yes, please, let's nuke them in a few versions 😄